programmatic_execution_of_notebooks/README.md

```diff
@@ -29,4 +29,82 @@ Details and limitations:
 you will need to go look for them in the execution directory of the workflow run.
 * This workflow was originally written for [app.terra.bio](https://app.terra.bio) and Google Cloud, but should run successfully on any Cromwell installation.
 * Environment variables `OWNER_EMAIL`, `WORKSPACE_BUCKET`, `WORKSPACE_NAME`, and `WORKSPACE_NAMESPACE` are not currently available, but this may change [in the future](https://www.google.com/url?q=https://support.terra.bio/hc/en-us/community/posts/4411972716443-Make-workspace-environment-variables-available-in-workflow-configuration&sa=D&source=docs&ust=1661812248047678&usg=AOvVaw0jzAJVDbmwco9I4jFIu85L). For now, if your notebook uses those, ensure that you can also inject the desired value via [Papermill](https://papermill.readthedocs.io/) parameters.
-* It is not compatible with notebooks written to run on [Hail](https://hail.is/) clusters.
\ No newline at end of file
+* It is not compatible with notebooks written to run on [Hail](https://hail.is/) clusters.
+
+## [dsub_notebook.py][./dsub_notebook.py]
+
+Use `dsub_notebook.py` to xecute Jupyter notebooks from the Terra Workspace terminal command line.  As with `notebook.wdl`,
+`dsub_notebook.py` executes Jupyter notebooks on a new, clean virtual machine independent of any local dependencies installed where you normally use
+Jupyter interactively.
+
+This script will:
+* Optionally install a list of Python packages before executing the notebook.
+  * This is because a kernel restart is often necessary to make use of Python packages installed during notebook execution time.
+  * For R package dependencies, have the notebook install them at the beginning.
+* Optionally pass parameters to the notebook via [Papermill](https://papermill.readthedocs.io/) to change its behavior.
+* Save the executed `ipynb` file containing cell outputs.
+
+Details and limitations:
+* If an error occurs during notebook execution, the resulting `ipynb` files are still saved, but
+you will need to go look for them in the output directory of the workflow run.
+
+## Demonstration notebook [DemoQCResultsWithParameters.ipynb](./DemoQCResultsWithParameters.ipynb)
+
+We have included a notebook to demonstrate how to programmatically execute a notebook using  `notebook.wdl` or
```

Comment:
These instructions are great. I think they could be refactored in a follow up PR in to a notebook that someone could run to setup dsub and the script for the more interesting example with sample QC. And then these instructions remaining in the README could be updated for the 'hello world' notebook to be included here.